module/reset_domain/include/mod_reset_domain.h

```diff
@@ -0,0 +1,217 @@
+/*
+ * Arm SCP/MCP Software
+ * Copyright (c) 2019-2020, Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Description:
+ *     Reset domain HAL.
+ */
+
+#ifndef MOD_RESET_DOMAIN_H
+#define MOD_RESET_DOMAIN_H
+
+#include <fwk_id.h>
+
+/*!
+ * \addtogroup GroupModules Modules
+ * \{
+ */
+
+/*!
+ * \defgroup GroupResetDomain Reset Domain HAL
+ *
+ * \details Support for setting the state of reset domains.
+ *
+ * \{
+ */
+
+/*!
+ * \brief API indices.
+ */
+enum mod_reset_domain_api_type {
+    MOD_RESET_DOMAIN_API_TYPE_HAL,
+    MOD_RESET_DOMAIN_API_TYPE_DRIVER_INPUT,
+    MOD_RESET_DOMAIN_API_COUNT,
+};
+
+/*!
+ * \brief Reset domain modes.
+ */
+enum mod_reset_domain_modes {
+    MOD_RESET_DOMAIN_AUTO_RESET = (1UL << 0),
+    MOD_RESET_DOMAIN_MODE_EXPLICIT_RESET = (1UL << 1),
+    MOD_RESET_DOMAIN_MODE_AUTO_RESET_ASYNC = (1UL << 2),
+};
+
+/*!
+ * \brief Reset domain states.
+ */
+enum mod_reset_domain_reset_states {
+    MOD_RESET_DOMAIN_RESET_STATE_ARCH_RESET = (0UL << 31),
+    MOD_RESET_DOMAIN_STATE_RESET_ID_COLD = (0UL << 0),
+};
+
+/*!
+ * \brief Reset domain capabilities.
+ */
+enum mod_reset_domain_capabilities {
+    MOD_RESET_DOMAIN_CAP_NOTIFICATION = (1UL << 31),
+    MOD_RESET_DOMAIN_CAP_ASYNC =  (1UL << 30)
+};
+
+/*!
+ * \brief Reset domain module configuration data.
+ *
+ */
+struct mod_reset_domain_config {
+    /*!
+     * \brief Notifications identifier.
+     */
+    fwk_id_t notification_id;
+
+};
+
+/*!
+ * \brief Reset domain element configuration data.
+ */
+struct mod_reset_domain_dev_config {
+    /*! Driver identifier */
+    fwk_id_t driver_id;
+
+    /*! Driver API identifier */
+    fwk_id_t driver_api_id;
+
+     /*! Supported modes (AUTORESET/AUTORESET ASYNC/EXPLICIT SIGNAL) */
+    uint32_t modes;
+
+    /*! Supported capabilities(ASYCN/NOTIFICATION SIGNAL) */
+    uint32_t capabilities;
+
+    /*! Supported reset states */
+    uint32_t reset_states;
```

Comment:
I've kept it reset_states as  uint32_t because there is a possibility of many states defined by the platform (see SCMI spec for reset_state encoding) which all can not be defined at this level. Hence keeping uint32_t is better IMO.